packages/flutter_tools/lib/src/commands/build_web.dart

```diff
@@ -67,6 +67,11 @@ class BuildWebCommand extends BuildSubCommand {
       help: 'Sets the optimization level used for Dart compilation to JavaScript. '
           'Valid values range from O0 to O4.'
     );
+    argParser.addOption(
+        'output-dir',
```

Comment:
1. can we match what `dart compile exe` does, which is have the option be named `--output` with a shortcut `-o`?
2. does this have to be an absolute path? can it be relative, like, `--ouput=../build/foo`?
3. are there any limitations to supporting this for every target platform?

Reply:
1 & 2: I was actually trying to follow the same pattern we had implemented for from `build_aar.dart` where they are using an absolute path with the same option definition. Should we instead follow what `dart compile exe` does instead?

3. I would need to do additional testing but I imagine if it has been working well in `build_aar.dart` and now web, perhaps we may be able to do it for every platform

Reply:
oh wow, we actually implemented this for build_aar?

I would say for any platform we can trivially implement this for we should. and if there are any that take more than a few lines of code, we should track implementing with a TODO and indicate from the option help text that it is not yet implemented.

Reply:
Following up from chat conversation; I will mark a TODO for implementing this feature across other build targets